cmd/main.go

```diff
@@ -37,7 +40,19 @@ var rootCmd = &cobra.Command{
 	Short:   "",
 	Long:    ``,
 	Run: func(cmd *cobra.Command, args []string) {
-		driver, err := hostpath.NewHostPathDriver(hostpath.DataRoot, hostpath.VolumeMapRoot, driverName, nodeID, endPoint, maxVolumesPerNode, version)
+		var kubeClient kubernetes.Interface
+		var err error
+
+		if readOnly {
+			fmt.Println("Read-Only mode: Loading Kubernetes client for HostPathDriver")
+
+			if kubeClient, err = loadKubeConfig(); err != nil {
```

Comment:
Aren't you basically just duplicating here what the `loadKubeConfig` function is doing? checking for an error etc. It would probably be cleaner to just drop the `loadKubeConfig` function and change this to
`if kubeClient, err = client.GetConfig(); err != nil {`

Reply:
The method is poorly named, it should reflect that in fact it loads the Kubernetes Config and then instantiate a `clientset`. Both actions can produce error, therefore the method holds them together. I'll rename it :-)